src/main/scala/com/github/drmarjanovic/app/api/routes/Articles.scala

```diff
@@ -0,0 +1,26 @@
+package com.github.drmarjanovic.app.api.routes
+
+import com.github.drmarjanovic.app.api.model.{ArticleResponse, ArticlesResponse, ErrorResponse}
+import com.github.drmarjanovic.app.domain.ArticlesRepo
+import zhttp.http._
+import zio.Has
+import zio.json._
+
+object Articles {
+  val routes: Http[Has[ArticlesRepo], Throwable, Request, UResponse] =
+    Http.collectM[Request] {
+      case req @ Method.GET -> Root / "articles" =>
+        val limit  = getQueryParam(req, "limit").getOrElse(20)
+        val offset = getQueryParam(req, "offset").getOrElse(0)
+
+        ArticlesRepo(_.all(limit, offset))
+          .map(data => ArticlesResponse.fromDomain(data).toJson)
+          .map(Response.jsonString)
+
+      case _ @Method.GET -> Root / "articles" / slug =>
+        ArticlesRepo(_.findBySlug(slug)).map {
+          case Some(article) => Response.jsonString(ArticleResponse.fromDomain(article).toJson)
+          case None          => unprocessableEntity(ErrorResponse.withReasons(s"Article $slug does not exist.").toJson)
```

Comment:
Why is it an unprocessable entity? :)

Reply:
I agree it's interesting. But [here](https://github.com/gothinkster/realworld/blob/main/api/swagger.json#L455) in the Swagger spec is defined that way.

Reply:
That's dumb, but well, if the spec says so :).